cmd/dao_tests/resource_test.go

```diff
@@ -0,0 +1,165 @@
+//go:build integration
+// +build integration
+
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	_ "github.com/RHEnVision/provisioning-backend/internal/dao/sqlx"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func createPkResource() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeNoop,
+		Handle:   "handle1",
+	}
+}
+
+func createPkResource2() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeAzure,
+		Handle:   "handle1",
+	}
+}
+
+func setupResource(t *testing.T) (dao.PubkeyResourceDao, context.Context) {
+	setup()
+	ctx := identity.WithTenant(t, context.Background())
+	resourceDao, err := dao.GetPubkeyResourceDao(ctx)
+	if err != nil {
+		panic(err)
+	}
+	return resourceDao, ctx
+}
+
+func teardownPubkeyResource(_ *testing.T) {
+	teardown()
+}
+
+func TestCreateResource(t *testing.T) {
+	resourceDao, ctx := setupResource(t)
+	defer teardownPubkeyResource(t)
+	resource := createPkResource()
+	err := resourceDao.Create(ctx, resource)
+	if err != nil {
+		panic(err)
```

Comment:
It is okay to use panic for GetXXXDao function because that is typically a configuration error. But for all other function calls, please fail the test properly so it will not stop the suite. You can take advantage that errors are also strings I believe, there is no reason to use formatted message and spend time writing "test failed blah blah":

```
		t.Fatal(err)
```

What is even better and I would highly encourage you to refactor not only this DAO test file, but all of them, is to use require package from the testify library we use:

```suggestion
		require.NoError(t, err)
```

This will immediately fail the test if there was an error effectively shortening the test code from three lines to one.